debug.py

```diff
@@ -84,7 +84,7 @@ def action_test(iterations: int = 1500, with_abort: bool = True):
         action.aborted()
 
 
-def to_upper(record: Record) -> Union[Record, List[Record], None]:
+def to_upper(record: Record) -> Record | List[Record] | None:
```

Comment:
Since we drop support for python > 3.10, it is safe to use built in types e.g `list[Record]` this would support most build in types. It is preferred to use generic types  by convention whenever possible.

Reply:
This is coming in my next PR. We decided to split changing to builtin types to a different PR to reduce the size of this PR.